compiler/modules/front/codeGen.bal

```diff
@@ -1090,15 +1089,18 @@ function codeGenExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Ex
                 return cx.semanticErr(`operand of ${"-"} must be int or float or decimal`, pos);
             }
             nextBlock.insns.push(insn);
-            return { result, block: nextBlock };
+            return codeGenNilLiftResult(cx, { result, block: nextBlock }, ifNilBlock, pos);
         }
         // Bitwise complement
         { opPos: var pos, op: "~",  operand: var o } => {
-            var { result: operand, block: nextBlock } = check codeGenExprForInt(cx, bb, env, o);
+            var { operand, nextBlock, ifNilBlock } = check codeGenUnaryNilLift(cx, env, o, bb, pos);
             bir:Register result = cx.createTmpRegister(t:INT, pos);
-            bir:IntBitwiseBinaryInsn insn = { op: "^", pos, operands: [-1, operand], result };
-            nextBlock.insns.push(insn);
-            return { result, block: nextBlock };
+            if operand is bir:IntOperand {
```

Comment:
Should have a function that takes an Operand and converts to an IntOperand|error.

Reply:
fixed